src/Device/Driver/XCVario.cpp

```diff
@@ -138,43 +174,98 @@ PXCV(NMEAInputLine &line, NMEAInfo &info)
     info.attitude.pitch_angle = Angle::Degrees(value);
   }
   // Space diagonal acceleration in X,Y,Z axes measure
-  if ( line.ReadChecked(x) && line.ReadChecked(y) && line.ReadChecked(z) )
+  if (line.ReadChecked(x) && line.ReadChecked(y) && line.ReadChecked(z))
     info.acceleration.ProvideGLoad(SpaceDiagonal(x, y, z));
 
   return true;
 }
 
+bool
+XVCDevice::XCV(NMEAInputLine &line, NMEAInfo &info)
+{
+  char topic[16];
+  line.Read(topic, sizeof(topic));
+  if (StringIsEqual(topic, "bal-water")) {
+    double value;
+    if (line.ReadChecked(value)) {
+      info.settings.ProvideBallastLitres(value, info.clock);
+    }
+  }
+  else if (StringIsEqual(topic, "version")) {
+    unsigned int value;
+    if (line.ReadChecked(value)) {
+      protocol_version = std::min(value, (unsigned int)XCV_VERSION_2);  // switch protocol version to the minimum version both can do
+    }
+  }
+  return true;
+}
+
+void XVCDevice::LinkTimeout()
+{
+  protocol_version = XCV_VERSION_UNKNOWN;   // on link timeout protocol negotiation is restarted
+  protocol_negotiation_timer = 5;                           // link timeout ticks = 5 seconds
+  xcvario_protocol_up = false;
+  nmea_enable = false;
+}
+
+bool XVCDevice::EnableNMEA([[maybe_unused]] OperationEnvironment &env)
+{
+  nmea_enable = true;
+  return true;
+}
+
 bool
 XVCDevice::ParseNMEA(const char *String, NMEAInfo &info)
 {
   if (!VerifyNMEAChecksum(String))
     return false;
-
   NMEAInputLine line(String);
   char type[16];
-  line.Read(type, 16);
-
-  if (StringIsEqual(type, "$PXCV"))
-    return PXCV(line, info);
-  else
-    return false;
+  line.Read(type, sizeof(type));
+  if (StringIsEqual(type, "$PXCV")) {                // cyclic data from device useful for channel supervision
+    xcvario_protocol_up = true;
+    if (protocol_version != XCV_VERSION_UNKNOWN) {   // only parse NMEA once protocol version is set
+      return PXCV(line, info);
+    }
+    return true;
+  }
+  else if (StringIsEqual(type, "!xcv")) {
+    return XCV(line, info);
+  }
+  return false;
 }
 
 // For documentation refer to chapter 10.1.3 Device Driver/XCVario in mulilingual handbook: https://xcvario.de/handbuch
+// Schedule sending of new protocol version 2; peer shall answer with the same message with its version
+// Upon reception of XCV's version, the protocol will be switched, atfer timeout we assume legacy version 1
+void
+XVCDevice::OnSysTicker()
+{
+  // repeat sending protocol version every second until timeout ( 5 seconds )
+  if (protocol_negotiation_timer && nmea_enable && protocol_version == XCV_VERSION_UNKNOWN && xcvario_protocol_up) {
+    protocol_negotiation_timer--;
+    if (protocol_version == XCV_VERSION_UNKNOWN) { // sent our version until negotiation is finished
```

Comment:
Another typo: `sent` -> `send` ("sent" is past tense)

Reply:
fixed